azurerm/internal/services/containers/kubernetes_nodepool.go

```diff
@@ -286,6 +298,14 @@ func FlattenDefaultNodePool(input *[]containerservice.ManagedClusterAgentPoolPro
 		name = *agentPool.Name
 	}
 
+	var nodeLabels map[string]string
+	if agentPool.NodeLabels != nil {
+		nodeLabels = make(map[string]string)
```

Comment:
Might be misreading here but can't we use your nifty flatten function here?

Reply:
`ExpandMapStringPtrString` is almost the same, but it accepts `map[string]interface{}`, so it can not be used here.